site/docs/generating_stardoc.md

```diff
@@ -0,0 +1,169 @@
+---
+layout: default
+title: Generating Documentation
+---
+
+<nav class="toc">
+  <h2>Contents</h2>
+  <ul>
+    <li><a href="#single-file">Single File</a></li>
+    <li><a href="#files-with-deps">Files with Dependencies</a></li>
+    <li><a href="#multiple-files">Multiple Files</a></li>
+  </ul>
+</nav>
+
+The following are some examples of how to use Stardoc.
+
+<a name="single-file"></a>
+## Single File
+
+Suppose you have a project containing Stardoc rules you want to document:
+
+```
+[workspace]/
+    WORKSPACE
+    checkstyle/
+        BUILD
+        checkstyle.bzl
+```
+
+To generate documentation for the rules in `checkstyle.bzl`, add the
+following target to `rules/BUILD`:
+
+```python
+load("@io_bazel_skydoc//skylark:stardoc.bzl", "stardoc")
+
+stardoc(
+    name = "checkstyle-docs",
+    input = "checkstyle.bzl",
+    out = "checkstyle_doc.md",
+)
+```
+
+Running `bazel build //checkstyle:checkstyle-docs` will generate a markdown file
+containing documentation for all Starlark rules defined in `checkstyle.bzl`.
+
+To generate a subset of rules defined in `checkstyle.bzl`, you may specify which
+rule names you specifically want documentation for using the `symbol_names` attribute
+of the `stardoc` rule. If `symbol_names` is specified, only rules matching a name
+in `symbol_names` will be documented:
```

Comment:
Do we have a use-case for `symbol_names`?

Reply:
Yes. A user may not want to generate documentation for everything in a single file (one of our internal users did request more granularity for generating rule documentation). This way, one can create multiple stardoc targets for generating separate rule documentation, even if the documentation is included in a single bzl file.

Also, this is very much required until we've completely migrated to the semantic change of not re-exporting loaded symbols. Without that, and without `symbol_names`, then even if we had a single bzl file which defined `foo_library` but it depended on a bzl file which contained `bar_library`, then both `foo_library` and `bar_library` would be generated in the same target.